src/WebJobs.Script.WebHost/WebHooks/WebJobsSdkExtensionHookProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Script.WebHost.Controllers;
+using HttpHandler = Microsoft.Azure.WebJobs.IAsyncConverter<System.Net.Http.HttpRequestMessage, System.Net.Http.HttpResponseMessage>;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost
+{
+    // Gives binding extensions access to a http handler. 
+    // This is registered with the JobHostConfiguration and extensions will call on it to register for a handler. 
+    internal class WebJobsSdkExtensionHookProvider : IWebhookProvider
+    {
+        // Map from an extension name to a http handler. 
+        private IDictionary<string, HttpHandler> _customHttpHandlers = new Dictionary<string, HttpHandler>(StringComparer.OrdinalIgnoreCase);
+
+        public IDictionary<string, HttpHandler> CustomHttpHandlers => _customHttpHandlers;
```

Comment:
nit: not sure if this is part of the IWebHookProvider interface, but rather than expose this data structure, wouldn't it be better to expose a GetHandlerOrNull(string name) method?